### DISCUSSION TITLE
Search commit by hash

### DISCUSSION THREAD

**Discussion by symonsg** [gitextensions/gitextensions#10608] [Q&A] [answered]
Hi,
since updating to version 4.0 I can't to search commit by hash.
Previously the search filter was "Commit message and hash" but now just "Commit message".
Will this feature be come back in the future?

Thanks,
Simone

**Comment by pmiossec** [comment]
Can you explain more your need?
What do you want to do exactly that is not covered by the "Go to commit..." navigation feature:
![image](https://user-images.githubusercontent.com/460196/210746488-a2aa0c28-699e-4f4c-889a-33be752747bb.png)

**Comment by gerhardol** [comment, marked as answer]
The following was removed, when filters were updated.
Use the navigate function instead. (then you do not have to clear the filter to see other than the commit itself).

![image](https://user-images.githubusercontent.com/6248932/210755829-150d98b9-c642-423c-a263-2d40748e87ee.png)

**Reply by 1Jesper1** [reply]
Thanks, didn't know there was another option.

**Comment by symonsg** [comment]
Hi @gerhardol, @pmiossec,
I didn't know that the function was removed when filters were updated.
Thanks I will use the "go to commit" feature

**Comment by starvational** [comment]
@gerhardol This was a very useful feature and I'm not sure why it was removed. I just spent time uninstalling and reinstalling v4.x and finally rolled back to 3.5.4 to find the missing **search by commit hash** feature.  

I bet I'm not the only dev who won't know to go into the navigate context menu or use some keyboard shortcut to invoke the search by commit hash feature.  Please reconsider adding it back to the main view ("quick filter") in the search box that we are all trained to use.

![image](https://user-images.githubusercontent.com/1291758/213513843-d7f7b129-e615-42bd-b29f-bd96069bf380.png)

update:  you noted "_then you do not have to clear the filter to see other than the commit itself_" as the reason

Just make it so that when I remove my search text, you update the view to clear the filter.  Currently I have to click on the GUI to get it to do this, which is an extra step, but can easily be fixed without removing the feature altogether.

**Reply by vbjay** [reply]
Click navigate menu. Go to commit.  Also explore the other options. You can literally do an incremental search. You can also use Ctrl f to do an advaned search.

**Comment by RussKie** [comment]
CTRL+SHIFT+G
